hpx/runtime/applier/applier.hpp

```diff
@@ -159,7 +159,8 @@ namespace hpx { namespace applier
         }
 
         /// Schedule  threads based on the given parcel
-        void schedule_action(parcelset::parcel p, std::size_t num_thread = -1);
+        void schedule_action(parcelset::parcel p, std::size_t num_thread =
+                             (std::numeric_limits<std::size_t>::max)());
```

Comment:
So, what are you trying to express here? Interpretation 1: num_thread should have a default value of -1. That's impossible because size_t is an unsigned type. Interpretation 2: num_thread should have the highest possible value. Then assigning -1 isn't exactly intuitive. 

IIRC -1 is an integer literal. Sign propagation rules and C++' weak typing for machine word types have the consequence that the result is the same. It's a hack my compiler complains about.